rootston/layer_shell.c

```diff
@@ -114,26 +114,58 @@ static void arrange_layers(struct wlr_output *_output) {
 	}
 }
 
-static void handle_destroy(struct wl_listener *listener, void *data) {
-	// TODO
+static void handle_output_destroy(struct wl_listener *listener, void *data) {
+	struct roots_layer_surface *layer =
+		wl_container_of(listener, layer, output_destroy);
+	layer->layer_surface->output = NULL;
+	wl_resource_destroy(layer->layer_surface->resource);
```

Comment:
This is dangerous, the next request on this object will disconnect the client.

Reply:
yeah TODO add close event to protocol

Reply:
This has been replaced with a close event